README.md
# expert-palm-tree

A repo for my code.


python-scripts/history.txt
    1  exit
    2  zsh --helpl
    3  zsh --help
    4  chsh -s $(which zsh)\n
    5  exit
    6  cd 
    7  cd ..
    8  ls
    9  cd ..
   10  ls
   11  cd media
   12  ls
   13  cd mzamayias
   14  ls
   15  cd Samsung
   16  ls Samsung
   17  sudo ls Samsung
   18  sudo cp Samsung/home.tar /home/mzamayias/Desktop
   19  sudo apt install rsync -y
   20  rsync --info=progress2 /Samsung/home.tar /home/mzamayias/Desktop
   21  rm /home/mzamayias/Desktop
   22  rm /home/mzamayias/Desktop/home.tar
   23  ls /home/mzamayias/Desktop
   24  rsync --info=progress2 Samsung/home.tar /home/mzamayias/Desktop
   25  sudo rsync --info=progress2 Samsung/home.tar /home/mzamayias/Desktop
   26  clear
   27  cd
   28  cd Desktop
   29  ls
   30  cd 
   31  cd Downloads
   32  ls
   33  bash ./Anaconda3-2020.02-Linux-x86_64.sh
   34  sudo apt install pigz -y
   35  clear
   36  tar -c --use-compress-program=pigz -f 
   37  tar --help
   38  tar -xc --use-compress-program=pigz -f home.tar /archive
   39  tar -c --use-compress-program=pigz -f home.tar /archive
   40  tar -xvcf home.tar /archive
   41  tar -xvcf home.tar 
   42  time pigz -dc home.tar| tar xf -
   43  ls
   44  rm -rf media
   45  pigz -dc home.tar | tar xf -
   46  clear
   47  ls
   48  rm -rf media
   49  tar -I pigz -xvf home.tar
   50  exit
   51  clear
   52  ls
   53  sudorm -rf media
   54  sudo rm -rf media
   55  cd
   56  cd ..
   57  cd media
   58  ls
   59  cd mz
   60  cd mzamayias
   61  LS
   62  ls
   63  cd Samsung
   64  cd
   65  clear
   66  cd Downloads
   67  sudo apt install ./codium_1.43.2-1585083818_amd64.deb -y
   68  exit
   69  λσ
   70  ls
   71  tar --use-compress-program=pigz home.tar /home/mzamayias/Desktop
   72  alias tar='tar --use-compress-program=pigz'\n
   73  tar -c /home/mzamayias/Desktop -xvf
   74  tar --help
   75  tar -c /home/mzamayias/Desktop -xf
   76  alias tar
   77  unalias tar
   78  alias tar
   79  clear
   80  tar -c /home/mzamayias/Desktop -xvf home.tar
   81  
   82  
   83  cd ..
   84  cd media/mzamayias/Samsung
   85  clear
   86  ls
   87  
   88  
   89  cd /media/mzamayias/Samsung
   90  tar --use-compress-program=pigz -xvf home.tar
   91  clear
   92  ls
   93  rm -rf media
   94  tar --use-compress-program="pigz --best --recursive | pv" -cf home.tar
   95  tar --use-compress-program="pigz -xf | pv" -cf home.tar
   96  tar --help
   97  tar --use-compress-program="pigz -x | pv" -cf home.tar
   98  tar --use-compress-program="pigz | pv" -xcf home.tar
   99  tar --use-compress-program="pigz | pv" -xf home.tar
  100  tar --use-compress-program="pigz --best --recursive | pv" -xf home.tar
  101  tar --use-compress-program="pigz --best | pv" -xf home.tar
  102  tar --use-compress-program="pigz | pv" -xf home.tar
  103  tar --use-compress-program=pigz -xf home.tar
  104  rm -rf media
  105  tar --use-compress-program=bzip2 -xf home.tar
  106  tar --use-compress-program=gzip2 -xf home.tar
  107  tar --use-compress-program=xf -xf home.tar
  108  tar --use-compress-program=pigz -xf home.tar
  109  clear
  110  rm -rf media
  111  time tar -xf home.tar /archive
  112  time tar --use-compress-program=pigz -xf home.tar
  113  python wallpaper.py
  114  lear
  115  clear
  116  python wallpaper.py
  117  codium wall
  118  codium wallpaper
  119  codium wallpaper.py
  120  python wallpaper.py
  121  pwd
  122  ls
  123  python wallpaper.py
  124  clear
  125  cd Desktop
  126  codium -n task2.py
  127  python task2.py
  128  sudo apt install steam-installer -y
  129  sudo codium -n /usr/share/themes/Materia-compact/gnome-shell/gnome-shell.css
  130  sudo codium -n /usr/share/themes/Materia-compact/gnome-shell/gnome-shell.css --user-data-dir
  131  clear
  132  exit
  133  sudo add-apt-repository ppa:cappelikan/ppa\n
  134  apt list --upgrable
  135  apt list --upgradable
  136  sudo apt update
  137  apt list --upgradable
  138  codium --version
  139  sudo apt-mark hold codium
  140  sudo apt-mark showhold
  141  sudo apt install mainline -y
  142  uname -r
  143  exot
  144  exit
  145  curl -sS https://download.spotify.com/debian/pubkey.gpg | sudo apt-key add - \n
  146  echo "deb http://repository.spotify.com stable non-free" | sudo tee /etc/apt/sources.list.d/spotify.list\n
  147  sudo apt update
  148  sudo apt install spotify-client -y
  149  clear
  150  cd Desktop
  151  ls
  152  mkdir time_tracker
  153  mv time_logger.py time_logger.py
  154  mv time_logger.py time_tracker
  155  cd time_tracker
  156  touch time_logs.csv
  157  codium -n .
  158  python time_logger.py
  159  clear
  160  python time_logger.py
  161  mv time_logs.csv time_logs.json
  162  codium time_log
  163  codium time_logs.json
  164  clear
  165  python time_logger.py
  166  ls
  167  clear
  168  python time_logger.py
  169  \tclear
  170  python 
  171  clear
  172  python
  173  lear
  174  clear
  175  python time_logger.py
  176  clear
  177  python time_logger.py
  178  clear
  179  python time_logger.py
  180  source /home/mzamayias/anaconda3/bin/activate
  181  conda activate base
  182  python time_logger.py
  183  clear && python time_logger.py
  184  python time_logger.py
  185  cleear
  186  clear
  187  cleear
  188  clear
  189  python time_logger.py
  190  cd Desktop
  191  ls
  192  cd Wallpapers
  193  ls
  194  python wallpaper.py
  195  codium -n .
  196  python wallpaper.py
  197  cd
  198  sudo apt install audacity -y
  199  sudo apt install transmission -y
  200  cd Audiobooks
  201  ls
  202  lame --ti atomic.jpg 'Atomic Habits Tiny Changes Remarkable Results.mp3'
  203  sudo apt install lame -y
  204  lame --ti atomic.jpg 'Atomic Habits Tiny Changes Remarkable Results.mp3'
  205  lame --ti indistractable.jpg 'Indistractable How to Control Your Attention and Choose Your Life.mp3'
  206  clear
  207  exit
  208  sudo apt install kid3 -y
  209  sudo apt remove --purge -y kid3
  210  sudo apt install -y easytag 
  211  sudo apt autoremove -y
  212  pwd
  213  ls
  214  exit

python-scripts/read_histories.py
Scriptsdef extend_list(small_list: list) -> list:
    big_list = []
    for package_name in small_list:
        big_list.extend(package_name)
    return big_list


def remove_from_list(list_to_clean: list, words_to_remove_to_clean: list) -> list:
    new_list = []
    for word in list_to_clean:
        if word not in new_list and word not in words_to_remove_to_clean:
            new_list.append(word)
    return new_list


def clean_list(list_to_clean: list) -> list:
    new_list = []Scripts
    for item in list_to_clean:
        if item not in new_list:
            new_list.append(item)
    return new_list


# read history
with open('history.txt', 'r', encoding='utf-8') as history:
    lines_1 = history.readlines()

# format lines
lines_1 = [
    line[7:-1]
    for line in lines_1
]

# read history
with open('zsh_history.txt', 'r', encoding='utf-8') as history:
    lines_2 = history.readlines()

# format lines
lines_2 = [
    line[15:-1]
    for line in lines_2
]

lines = lines_1 + lines_2

# list of packages to install
packages_to_install = []

# list of packages to remove
packages_to_remove = []

# words_to_remove to remove
words_to_remove = [
    'sudo', 'apt', 'install', 'remove', '-y', '--fix-broken', 'albert',
    'solaar-gnome3', 'piper', 'python3-pyudev', 'cups-pdf', '-y\\', 'apt-get',
    'ghostscript', 'unoconv', 'libreoffice-java-common', 'sudo', 'libvulkan1',
    'mesa-vulkan-drivers', 'mesa-vulkan-drivers', 'libratbag', 'glxinfo',
    'mesa-utils', 'wv', '\x1b[200~sudo', 'vulkan-utils', '', 'build-essential',
    'kernel-package', 'fakeroot', 'libssl-dev', 'ccache', 'flex', 'bison',
    'libelf-dev', 'libncurses5-dev', 'wget', 'cmake', 'python3', 'python-minimal',
    'python-mako', 'python3-mako', 'pkg-config', 'libexpat1-dev', 'libxrandr-dev',
    'libdrm-dev', 'libxdamage-dev', 'libxcb-glx0-dev', 'libxcb-dri2-0-dev',
    'libxcb-dri3-dev', 'libxcb-present-dev', 'libxshmfence-dev', 'libxxf86vm-dev',
    'libx11-xcb-dev', 'ninja-build', 'inkscape', 'steam-installer-y', 'clang',
    'libgtk-3-dev', 'libblkid-dev', 'liblzma-dev', 'pm-utils', 'hibernate',
    'lutris', 'nvidia-driver-455', 'nvidia-driver-450', 'nvidia-driver-435',
    'nvidia-graphics-drivers-455', 'lv2-dev', 'meson', 'fftw-dev', 'lv2',
    '-ylv2-dev', "'^lv2*'", 'python3.8-dev', 'pop-desktop-y', 'pop-desktop',
    'openjdk-8', 'openjdk-8-jdk', 'ttf-mscorefonts-installer', 'fonts-firacode',
    'dart', '--purge', 'steam', 'autoremove', 'solaar', '-y;', 'autoclean', '--purge\\',
    'printer-driver-cups-pdf', '\'^libreoffice.*\'--purge', '--allow-remove-essential',
    'reboot', 'slack-desktop', '&&', 'upgrade', 'update', './codium_1.43.2-1585083818_amd64.deb',
    'spotify-client', 'slack', 'inkscape-y', 'gnome-shell-extension-ubuntu-dock',
    'libc6:i386', 'libncurses5:i386', 'libstdc++6:i386', 'lib32z1', 'libbz2-1.0:i386', 'lm-sensors',
    'debhelper', 'dbus', 'google-mock', 'libboost-dev', 'libboost-filesystem-dev', 'libboost-log-dev',
    'libboost-iostreams-dev', 'libboost-program-options-dev', 'libboost-system-dev libboost-test-dev',
    'libboost-thread-dev', 'libcap-dev libsystemd-dev', 'libegl1-mesa-dev', 'libgles2-mesa-dev',
    'libglm-dev', 'libgtest-dev', 'liblxc1', 'libproperties-cpp-dev', 'libprotobuf-dev', 'libsdl2-dev',
    'libsdl2-image-dev', 'lxc-dev', 'protobuf-compiler', 'anbox', 'lame', 'kid3', 'easytag', 'mainline',
    'cmake-data', '\\\\n', 'libboost-system-dev', 'libboost-test-dev', 'libcap-dev', 'libsystemd-dev',
    'python3-setuptools', 'python3-gi', 'python-gobject'

]

# populate lists
for line in lines:
    if 'sudo' in line and 'apt' in line:
        if 'install' in line:
            if line not in packages_to_install:
                packages_to_install.append(line)
        if 'remove' in line:
            if line not in packages_to_remove:
                packages_to_remove.append(line)

packages_to_install = [
    package_name.split(' ')
    for package_name in packages_to_install
]
packages_to_install = remove_from_list(clean_list(extend_list(packages_to_install)), words_to_remove)

packages_to_remove = [
    package_name.split(' ')
    for package_name in packages_to_remove
]
packages_to_remove = remove_from_list(clean_list(extend_list(packages_to_remove)), words_to_remove)
